cas-server-documentation/integration/Delegate-Authentication.md

```diff
@@ -287,7 +245,22 @@ To start authentication on a remote provider, these links must be added on the l
 </form>
 {% endhighlight %}
 
+
 ##Demo
 
 Take a look at this demo: [cas-pac4j-oauth-demo](https://github.com/leleuj/cas-pac4j-oauth-demo) to see this authentication delegation mechanism in action.
 
+
+##Wrapper for pac4j authenticator and profile creator
+
+It's another feature: the `cas-server-support-pac4j` module has a wrapper to use a pac4j authenticator and profile creator as a CAS authentication handler.
+
+It has a first implementation for username/password credentials: `org.jasig.cas.support.pac4j.authentication.handler.support.UsernamePasswordWrapperAuthenticationHandler` (extended from `AbstractWrapperAuthenticationHandler`):
+
+{% highlight xml %}
+<bean id="simpleAuthenticator" class="org.pac4j.http.credentials.authenticator.test.SimpleTestUsernamePasswordAuthenticator" />
+
+<bean id="primaryAuthenticationHandler" class="org.jasig.cas.support.pac4j.authentication.handler.support.UsernamePasswordWrapperAuthenticationHandler">
```

Comment:
Understood. At the same time, I think this will force us to create individual authN handlers for everything that pac4j offers. Not sure that's a good thing, but I understand the problem and your solution makes sense.